specifyweb/specify/views.py

```diff
@@ -555,5 +583,85 @@ def record_merge(request: http.HttpRequest, model_name: str, old_model_id: int,
     # if model_name.lower() != 'Agent'.lower():
     #     return http.HttpResponseForbidden(f'Record Merging for the {model_name} table is not yet been fully tested.')
 
-    response = record_merge_fx(model_name, old_model_id, new_model_id)
+    response = record_merge_fx(model_name, [old_model_id], new_model_id)
+    return response
+
+@openapi(schema={
+    'post': {
+        "requestBody": {
+            "required": True,
+            "description": "Rplace a list of old records with a new record.",
+            "content": {
+                "application/json": {
+                    "schema": {
+                        "type": "object",
+                        "description": "The request body.",
+                        "properties": {
+                            "model_name": {
+                                "type": "string",
+                                "description": "The name of the table that is to be merged."
+                            },
+                            "new_model_id": {
+                                "type": "integer",
+                                "description": "The new ID value of the model that is replacing the old one."
+                            },
+                            "old_record_ids": {
+                                "type": "array",
+                                "items": {
+                                    "type": "integer"
+                                },
+                                "description": "The old record IDs."
+                            },
+                            "new_record_data": {
+                                "type": "object",
+                                "description": "The new record data."
+                            }
+                        },
+                        'required': ['model_name', 'new_model_id', 'collection_id', 'old_record_ids', 'new_record_data'],
+                        'additionalProperties': False
+                    }
+                }
+            }
+        },
+        "responses": {
+            "204": {"description": "Success",},
+            "404": {"description": "The ID specified does not exist."},
+            "405": {"description": "A database rule was broken."}
+        }
+    },
+})
+
+@login_maybe_required
+@require_POST
+def record_merge_extended(request: http.HttpRequest, model_name: str, new_model_id: int) -> http.HttpResponse:
+    """Replaces all the foreign keys referencing the old record IDs
+    with the new record ID, and deletes the old records.
+    """
+    request_params = http.QueryDict(request.META['QUERY_STRING'])
```

Comment:
there should be an easier way to do this.
Something like this:
```py
if 'version' in request.GET:
  version = request.GET['version']
else:
  version = getattr(spmodels, model_name.title()).objects.get(id=new_model_id).version
```

or maybe even this, to avoid needless if/else:
```python
default_version = getattr(spmodels, model_name.title()).objects.get(id=new_model_id).version
version = request.GET.get('version', default_version)
```